Moq.AutoMock/AutoMocker.cs

```diff
@@ -325,7 +325,7 @@ internal object CreateInstanceInternal(Type type, ObjectGraphContext context)
             typeMap[typeof(TImplementation)] = new MockInstance(selfMock);
             typeMap[typeof(TService)] = new MockInstance(selfMock.As<TService>());
         });
-        return selfMock.Object;
+        return selfMock.As<TService>();

```

Comment:
Rather than re-calling `selfMock.As<TService>()` twice, let's pull this up into a local variable and use that.

Reply:
Fixed